### PR TITLE
`Pkg.status` improvements

### PR DESCRIPTION
**Pull request by 00vareladavid**
A possible alternative to #1656.

This attempts to rectify the concerns in #1621. 

Some example output:
```
(demo0) pkg> status
     Status `/tmp/demo0/Project.toml` (empty project)

(demo0) pkg> status --manifest
     Status `/tmp/demo0/Manifest.toml` (empty manifest)

(demo0) pkg> status --diff
 No Changes to `/tmp/demo0/Project.toml`

(demo0) pkg> status --diff --manifest
 No Changes to `/tmp/demo0/Manifest.toml`

(demo0) pkg> add Example
  Resolving package versions...
   Updating `/tmp/demo0/Project.toml`
  [7876af07] + Example v0.5.3
   Updating `/tmp/demo0/Manifest.toml`
  [7876af07] + Example v0.5.3

(demo0) pkg> add Example
  Resolving package versions...
 No Changes to `/tmp/demo0/Project.toml`
 No Changes to `/tmp/demo0/Manifest.toml`

(demo0) pkg> status
     Status `/tmp/demo0/Project.toml`
  [7876af07] Example v0.5.3

(demo0) pkg> status --diff
       Diff `/tmp/demo0/Project.toml`
  [7876af07] + Example v0.5.3

(demo0) pkg> status --diff Foo
 No Matches in diff for `/tmp/demo0/Project.toml`

(demo0) pkg> status --diff Example
       Diff `/tmp/demo0/Project.toml`
  [7876af07] + Example v0.5.3

(demo0) pkg> status --diff --manifest Example
       Diff `/tmp/demo0/Manifest.toml`
  [7876af07] + Example v0.5.3

(demo0) pkg> add Markdown
  Resolving package versions...
   Updating `/tmp/demo0/Project.toml`
  [d6f4376e] + Markdown
   Updating `/tmp/demo0/Manifest.toml`
  [2a0f44e3] + Base64
  [d6f4376e] + Markdown

julia> push!(empty!(DEPOT_PATH), "/tmp/depot")
1-element Array{String,1}:
 "/tmp/depot"

(demo0) pkg> status
     Status `/tmp/demo0/Project.toml`
→ [7876af07] Example v0.5.3
  [d6f4376e] Markdown
Info packages marked with → are not downloaded, use `instantiate` to download
```

### CODE REVIEW

**Comment by 00vareladavid** [comment]
For reference here is a comparison of `status` printing before this and #1621:

```
(demo2) pkg> status
Status `/tmp/demo2/Project.toml`
  (no changes since last commit)

(demo2) pkg> status --manifest
Status `/tmp/demo2/Manifest.toml`
  (no changes since last commit)

(demo2) pkg> status --diff
Status `/tmp/demo2/Project.toml`
  (no changes since last commit)

(demo2) pkg> status --diff --manifest
Status `/tmp/demo2/Manifest.toml`
  (no changes since last commit)

(demo2) pkg> add Example
  Resolving package versions...
   Updating `/tmp/demo2/Project.toml`
  [7876af07] + Example v0.5.3
   Updating `/tmp/demo2/Manifest.toml`
  [7876af07] + Example v0.5.3

(demo2) pkg> add Example
  Resolving package versions...
   Updating `/tmp/demo2/Project.toml`
 [no changes]
   Updating `/tmp/demo2/Manifest.toml`
 [no changes]

(demo2) pkg> status --diff
Status `/tmp/demo2/Project.toml`
  [7876af07] + Example v0.5.3

(demo2) pkg> status --diff Foo
ERROR: The following package names could not be resolved:
 * Foo (not found in project or manifest)


(demo2) pkg> status --diff Example
Status `/tmp/demo2/Project.toml`
  [7876af07] + Example v0.5.3

(demo2) pkg> status --diff --manifest Example
Status `/tmp/demo2/Manifest.toml`
  [7876af07] + Example v0.5.3

(demo2) pkg> add Markdown
  Resolving package versions...
   Updating `/tmp/demo2/Project.toml`
  [d6f4376e] + Markdown 
   Updating `/tmp/demo2/Manifest.toml`
  [2a0f44e3] + Base64 
  [d6f4376e] + Markdown 

julia> push!(empty!(DEPOT_PATH), "/tmp/depot")
1-element Array{String,1}:
 "/tmp/depot"

(demo2) pkg> status
Status `/tmp/demo2/Project.toml`
→ [7876af07] Example v0.5.3
  [d6f4376e] Markdown 
┌ Warning: Some packages (indicated with a red arrow) are not downloaded, use `instantiate` to instantiate the current environment
└ @ Pkg.Display ~/pkg/pkg_master/src/Display.jl:238
```

**Comment by KristofferC** [comment]
`printpkgstatus` has an optional last argument to turn off indentation. I think it is good to use this for e.g. `Status` and `Diff`.

File: src/Operations.jl [hunk 1]
```diff
@@ -1582,45 +1582,56 @@ function diff_array(old_ctx::Union{Context,Nothing}, new_ctx::Context; manifest=
...
+        all_packages_downloaded &= pkg_downloaded
+        printstyled(ctx.io, pkg_downloaded ? " " : "→"; color = :red)
+        printstyled(ctx.io, " [", string(uuid)[1:8], "] "; color = :light_black)
         diff ? print_diff(ctx, old, new) : print_single(ctx, new)
         println(ctx.io)
     end
-    if !printed_something
-        printpkgstyle(ctx, header, (diff && empty_diff) ? "empty diff" : "no matches")
+    if !all_packages_downloaded
+        printpkgstyle(ctx, :Info, "packages marked with → are not downloaded, use `instantiate` to download", true)
```
Comment (KristofferC) [inline src/Operations.jl]: I think there is some extra clarity to do it the way I did in https://github.com/JuliaLang/Pkg.jl/pull/1656/files#diff-920ed338dcbd9d62489c6d12dc2fbb18R1636 to show the arrow as red.

Comment (00vareladavid) [reply src/Operations.jl]: ah, yes that is definitely better

**Comment by 00vareladavid** [comment]
K, I'll turn off indentation as well

**Comment by KristofferC** [comment]
Pushed the dicsussed changes here.

**Pull request merged**